src/AWS.Deploy.Orchestration/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,222 @@
+        private readonly string _targetApplicationFullPath;
+
+        private const string DEPLOYMENT_MANIFEST_FILE_NAME = "aws-deployments.json";
+
+        public DeploymentManifestEngine(IDirectoryManager directoryManager, IFileManager fileManager, OrchestratorSession session, string targetApplicationFullPath)
+        {
+            _directoryManager = directoryManager;
+            _fileManager = fileManager;
+            _session = session;
+            _targetApplicationFullPath = targetApplicationFullPath;
+        }
+
+        /// <summary>
+        /// This method updates the deployment manifest json file by adding the directory path at which the CDK deployment project is saved.
+        /// If the manifest file does not exists then a new file is generated.
+        /// <param name="saveCdkDirectoryFullPath">The absolute path to the directory at which the CDK deployment project is saved</param>
+        /// <exception cref="FailedToUpdateDeploymentManifestFileException">Thrown if an error occured while trying to update the deployment manifest file.</exception>
+        /// </summary>
+        public async Task UpdateSaveCdkProject(string saveCdkDirectoryFullPath)
+        {
+            try
+            {
+                if (!_directoryManager.Exists(saveCdkDirectoryFullPath))
+                    return;
+
+                var deploymentManifestModel = await GetDeploymentManifest();
+                var targetApplicationDirectoryPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+                var saveCdkDirectoryRelativePath = _directoryManager.GetRelativePath(targetApplicationDirectoryPath, saveCdkDirectoryFullPath);
+
+                if (deploymentManifestModel != null)
+                {
+                    deploymentManifestModel.DeploymentManifestEntries.Add(new DeploymentManifestEntry(saveCdkDirectoryRelativePath));
+                }
+                else
+                {
+                    var deploymentManifestEntries = new List<DeploymentManifestEntry> { new DeploymentManifestEntry(saveCdkDirectoryRelativePath) };
+                    deploymentManifestModel = new DeploymentManifestModel(new(), deploymentManifestEntries);
+                }
+
+                await WriteDeploymentManifestFile(deploymentManifestModel);
+            }
+            catch (Exception ex)
+            {
+                throw new FailedToUpdateDeploymentManifestFileException($"Failed to update the deployment manifest file " +
+                    $"for the deployment project stored at '{saveCdkDirectoryFullPath}'", ex);
+            }
+            
+        }
+
+        /// <summary>
+        /// This method updates the deployment manifest json file by adding the name of the stack that was most recently used.
+        /// If the manifest file does not exists then a new file is generated.
+        /// </summary>
+        public async Task UpdateLastDeployedStack(string stackName)
```

Comment:
Should this method also clean up any orphan stacks in the LastDeployedStacks for the current account and region? Or is there some other mechanism to clean up the file for stacks deleted in the console or some other tool.